mandatory/2-function-creation.js

```diff
@@ -15,26 +21,60 @@ function isAcceptableUser(userAge, isLoggedIn) {}
     is applieds and 142.5 should be returned)
 */
 
-function applyDiscount(totalPrice) {}
+function applyDiscount(totalPrice) {
+  if (totalPrice > 200) {
+    return 0.9 * totalPrice;
+  } else if (totalPrice > 0 && totalPrice < 200) {
+    return 0.95 * totalPrice;
+  } else {
+    return 0
+  }
+}
 
 /*
   Complete the function to print to the console the odd numbers between 1 and limit (use a while loop):
   */
-function printOddNumbers(limit) {}
+
+
+  function printOddNumbers(limit) {
+    let i = 0;
+    while ( i <= limit) {
+      if (i % 2 === 1) {
+        console.log(i);
+      } 
+      i = i + 1;
+    }
+  }
 
 /*
   Complete the buyTwoGetTheCheapestFree function: if user buys two items, the cheapest item will be free!
   The function should return the price to be paid once the discount is applied.
 */
-function buyTwoGetTheCheapestFree(price1, price2) {}
+function buyTwoGetTheCheapestFree(price1, price2) {
+  if (price1 > price2) {
+    return price1;
+  } else if (price2 > price1) {
+    return price2;
+  } else price1;
+}
 
 /*
   Complete the function to determine if it is suitable for a person to register based on their age!
   - if the person is 12 or younger it should return "You Are Too Young To Register"
   - if the person is older than 12 and younger than 90 it should return "You Can Register"
   - if the person is 90 or older it should return "You Don't Need To Register"
 */
-function canRegister(age) {}
+function canRegister(age) {
+  if (age <= 12) {
+    return "You Are Too Young To Register";
+  } else if (age > 12 && age < 90) {
+    return "You Can Register";
+  } else if (age >= 90) {
+    return "You Don't Need To Register";
+  } else {
```

Comment:
The last statement that returns false is not required. recommend to delete the last else statement :) 

```
function canRegister(age) {
  if (age <= 12) {
    return "You Are Too Young To Register";
  } else if (age > 12 && age < 90) {
    return "You Can Register";
  } else {
    return "You Don't Need To Register";
```